mempool/reactor_test.go

```diff
@@ -685,3 +685,184 @@ func TestMempoolVectors(t *testing.T) {
 		require.Equal(t, tc.expBytes, hex.EncodeToString(bz), tc.testName)
 	}
 }
+
+// Verify that counting of duplicates and first time transactions work
+// The test sends transactions from node2 to node1 twice.
+// The second time they will get rejected.
+func TestDOGTransactionCount(t *testing.T) {
+	config := cfg.TestConfig()
+	config.Mempool.DOGProtocolEnabled = true
+
+	// Put the interval to a higher value to make sure the values don't get reset
+	config.Mempool.DOGAdjustInterval = 15 * time.Second
+	reactors, _ := makeAndConnectReactors(config, 2, nil)
+
+	// create random transactions
+	txs := NewRandomTxs(numTxs, 20)
+	secondNodeID := reactors[1].Switch.NodeInfo().ID()
+	secondNode := reactors[0].Switch.Peers().Get(secondNodeID)
+
+	txUnique := make(map[string][]byte, len(txs))
+	for _, tx := range txs {
+		txUnique[string(tx.Hash())] = tx
+	}
+
+	for _, tx := range txUnique {
+		_, err := reactors[0].TryAddTx(tx, secondNode)
+		require.NoError(t, err)
+	}
+
+	require.Equal(t, int64(len(txUnique)), reactors[0].redundancyControl.firstTimeTxs)
+	for _, tx := range txUnique {
+		_, err := reactors[0].TryAddTx(tx, secondNode)
+		// The transaction is in cache, hence the Error
+		require.Error(t, err)
+	}
+	require.Equal(t, int64(len(txUnique)), reactors[0].redundancyControl.duplicateTxs)
+
+	reactors[0].redundancyControl.triggerAdjustment()
+	time.Sleep(1 * time.Second)
```

Comment:
This can be replaced by `reactors[0].redundancyControl.adjustRedundancy(reactors[0])`.

Reply:
I would leave `triggerAdjustments` as is because the function should then call `adjustRedundancy` followed by a `Reset`.

Reply:
Ok. Sounds good. I guess the 1 second delay is to wait for reactors[0] to send a message and for reactors[2] to disable a route. Is 100 ms enough for that? Waiting one second here and there slow downs a lot the tests.

Reply:
I cancheck,maybe it will be enough

Reply:
Lowered the value to 100ms.